recipes/opusfile/all/conanfile.py

```diff
@@ -0,0 +1,90 @@
+
+    @property
+    def _is_msvc(self):
+        return self.settings.os == "Windows" and \
+               self.settings.compiler == "Visual Studio"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self._is_msvc and self.options.shared:
+            raise ConanInvalidConfiguration("Opusfile doesn't support building as shared with Visual Studio")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _build_vs(self):
+        includedir = os.path.abspath(os.path.join(self._source_subfolder, "include"))
+        with tools.chdir(os.path.join(self._source_subfolder, "win32", "VS2015")):
+            msbuild = MSBuild(self)
+            msbuild.build_env.include_paths.append(includedir)
+            msbuild.build(project_file="opusfile.sln", targets=["opusfile"],
+                          platforms={"x86": "Win32"})
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            with tools.chdir(self._source_subfolder):
+                self.run("./autogen.sh", win_bash=tools.os_info.is_windows)
+            args = []
+            if self.options.shared:
+                args.extend(["--disable-static", "--enable-shared"])
+            else:
+                args.extend(["--disable-shared", "--enable-static"])
+            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        if self._is_msvc:
+            self._build_vs()
+        else:
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        if self._is_msvc:
+            include_folder = os.path.join(self._source_subfolder, "include")
+            self.copy(pattern="*", dst=os.path.join("include", "opus"), src=include_folder)
+            self.copy(pattern="*.dll", dst="bin", keep_path=False)
+            self.copy(pattern="*.lib", dst="lib", keep_path=False)
+        else:
+            autotools = self._configure_autotools()
+            autotools.install()
+            tools.rmdir(os.path.join(self.package_folder, "share"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            os.remove(os.path.join(self.package_folder, "lib", "libopusfile.la"))
+            os.remove(os.path.join(self.package_folder, "lib", "libopusurl.la"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["opusfile"]
+        self.cpp_info.includedirs.append(os.path.join("include", "opus"))
```

Comment:
```suggestion
        self.cpp_info.includedirs.append(os.path.join("include", "opus"))
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["m", "dl", "pthread"]
```

ldd lib/libopusfile.so
	linux-vdso.so.1 (0x00007ffe8c5ed000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f843d382000)
	libdl.so.2 => /usr/lib/libdl.so.2 (0x00007f843d37d000)
	libpthread.so.0 => /usr/lib/libpthread.so.0 (0x00007f843d35b000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f843d194000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f843d55d000)